src/main/java/edu/phema/elm_to_omop/vocabulary/ConceptCodeCsvFileValuesetService.java

```diff
@@ -9,39 +9,114 @@
 import org.ohdsi.circe.vocabulary.Concept;
 import org.ohdsi.circe.vocabulary.ConceptSetExpression;
 
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.InputStream;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class ConceptCodeCsvFileValuesetService implements IValuesetService {
     private IOmopRepositoryService omopService;
 
-    private String csvFilePath;
-    private String valuesetSpreadsheetTab;
+    private String csvPath;
+    private int conceptSetId;
 
-    public ConceptCodeCsvFileValuesetService(IOmopRepositoryService omopService, String csvFilePath, String valuesetSpreadsheetTab) {
+    public ConceptCodeCsvFileValuesetService(IOmopRepositoryService omopService, String csvPath) {
         this.omopService = omopService;
-        this.csvFilePath = csvFilePath;
-        this.valuesetSpreadsheetTab = valuesetSpreadsheetTab;
+        this.csvPath = csvPath;
+
+        this.conceptSetId = 0;
     }
 
     @Override
     public PhemaConceptSetList getConceptSetList() throws ValuesetServiceException {
+        if (this.isDirectory()) {
+            List<String> csvFiles = getCsvFilesInDirectory(csvPath);
+
+            List<PhemaConceptSetList> conceptSetListList = new ArrayList<>();
+            for (String csvFilename : csvFiles) {
+                conceptSetListList.add(this.getConceptSet(csvFilename));
+            }
+
+            // Flatten the result into a single PhemaConceptSetList
+            PhemaConceptSetList result = new PhemaConceptSetList();
+            for (PhemaConceptSetList list : conceptSetListList) {
+                result.addAllConceptSets(list.getConceptSets());
+                result.addAllNotFoundCodes(list.getNotFoundCodes());
+            }
+
+            return result;
+        } else {
+            return this.getConceptSet(this.csvPath);
+        }
+    }
+
+    private List<String> getCsvFilesInDirectory(String directoryPath) {
+        File directory;
+
+        if (this.isResource()) {
+            directory = new File(this.getClass().getResource(csvPath).getPath());
+        } else {
+            directory = new File(directoryPath);
+        }
+
+        File[] csvFiles = directory.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.toUpperCase().endsWith(".CSV");
```

Comment:
I was going to suggest equalsIgnoreCase here, but after some research there doesn't seem to be a huge efficiency gain, and it may work differently across locales.  TL;DR - nothing needs to be changed, I'm commenting so you know I did actually read the code ;)